### PR TITLE
feat: Add Next.js example of linear file uploading

### PR DESCRIPTION
**Pull request by pacocoursey**
Adds a new top-level `examples/` folder for public examples of Linear API/SDK features. The first guide added here is an example of how to upload a file to Linear using Next.js API Routes.

### CODE REVIEW

**Comment by guillaumelachaud** [comment]
@pacocoursey Did you manage to get the SDK up and running in a Next.JS middleware? We have an open issue to make it compatible: #349

**Comment by pacocoursey** [comment]
> @pacocoursey Did you manage to get the SDK up and running in a Next.JS middleware? We have an open issue to make it compatible: https://github.com/linear/linear/issues/349

@guillaumelachaud No, this is an unrelated change that does not affect the SDK.

I can take a look at that middleware issue at some point, though!

File: examples/nextjs-file-upload/pages/api/uploadFileToLinear.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  try {
+    await fetch(uploadUrl, {
+      method: "PUT",
+      body: file.buffer,
+      headers,
+    });
+
+    return assetUrl;
+  } catch (e) {
+    throw new Error("Failed to upload file to Linear", e);
```
Comment (tommoor) [inline examples/nextjs-file-upload/pages/api/uploadFileToLinear.ts:58]: Build error here seems correct, should use string interpolation.

File: examples/nextjs-file-upload/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Upload a file to Linear with Next.js
+
+This example shows how to upload a file to Linear using the Linear TypeScript SDK and Next.js API Routes. To run this example, you'll need a Linear account and a Linear API key. You can create an API key from your personal [Linear settings](https://linear.app/settings/api). Learn more about authentication in the [Linear Developer documentation](https://developers.linear.app/docs/sdk/getting-started#2.-create-a-linear-client).
```
Comment (tommoor) [inline examples/nextjs-file-upload/README.md:3]: Should we be more explicit that while this example is in Next.JS the principals apply to any Node backend?

File: examples/nextjs-file-upload/pages/api/uploadFileToLinear.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  if (!uploadPayload.success || !uploadPayload.uploadFile) {
+    throw new Error("Failed to request upload URL");
+  }
+
+  const uploadUrl = uploadPayload.uploadFile.uploadUrl;
+  const assetUrl = uploadPayload.uploadFile.assetUrl;
+
+  // Make sure to copy the response headers for the PUT request
+  const headers = new Headers();
+  headers.set("Content-Type", file.mimetype);
```
Comment (tommoor) [inline examples/nextjs-file-upload/pages/api/uploadFileToLinear.ts:45]:
```suggestion

  // It is important that the content-type of the request matches the value passed as the first argument to `fileUpload`.
  headers.set("Content-Type", file.mimetype);
```

File: examples/nextjs-file-upload/pages/api/uploadFileToLinear.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  headers.set("Cache-Control", "public, max-age=31536000");
+  uploadPayload.uploadFile.headers.forEach(({ key, value }) => headers.set(key, value));
+
+  try {
+    await fetch(uploadUrl, {
+      method: "PUT",
```
Comment (tommoor) [inline examples/nextjs-file-upload/pages/api/uploadFileToLinear.ts:51]:
```suggestion
      // Note PUT is important here, other verbs will not work.
      method: "PUT",
```

**Pull request merged**